Maple2.Server.Game/Manager/Field/FieldManager.State.cs

```diff
@@ -192,6 +203,30 @@ public partial class FieldManager {
         return fieldInteract;
     }
 
+    public FieldInteract? AddInteract(InteractObject interactData, IInteractObject interactObject) {
```

Comment:
What about combining these functions:

```cs
public FieldInteract? AddInteract(string entityId, InteractObject interact) {
    IInteractObject interactObject = interact switch {
        Ms2InteractMesh mesh => new InteractMeshObject(entityId, mesh),
        Ms2Telescope telescope => new InteractTelescopeObject(entityId, telescope),
        Ms2SimpleUiObject ui => new InteractUiObject(entityId, ui),
        Ms2InteractDisplay display when metadata.Type == InteractType.DisplayImage => new InteractDisplayImage(entityId, display),
        Ms2InteractDisplay poster when metadata.Type == InteractType.GuildPoster => new InteractGuildPosterObject(entityId, poster),
        Ms2InteractActor actor => new InteractGatheringObject(entityId, actor),
        _ => throw new ArgumentException($"Unsupported Type: {metadata.Type}"),
    };

    return AddInteract(interact, interactObject);
}

public FieldInteract? AddInteract(InteractObject interactData, IInteractObject interactObject) {
    if (!TableMetadata.InteractObjectTable.Entries.TryGetValue(interactData.InteractId, out InteractObjectMetadata? metadata)) {
        return null;
    }

    var fieldInteract = new FieldInteract(this, NextLocalId(), interactObject.EntityId, metadata, interactObject) {
        Position = interactData.Position,
        Rotation = interactData.Rotation,
    };

    //TODO: Add treasure chests
    switch (interactObject) {
        case InteractBillBoardObject billboard:
            fieldAdBalloons[billboard.EntityId] = fieldInteract;
            break;
        default:
            fieldInteracts[interactObject.EntityId] = fieldInteract;
            break;
    }

    return fieldInteract;
}
```


`Broadcast(InteractObjectPacket.Add(interactObject));` should be sent separately for consistency with all the other Add functions.

Reply:
The first `AddInteract` method would need the metadata. This could work if we get the metadata in it and remove it from the second one. This would just mean I would need to change the parameters to include metadata (and then looking up and providing the metadata from `ItemUseHandler` for Ad Balloons if you're good with it?

Reply:
Oh I see, could you just fetch Metadata twice? I guess it doesn't matter all that much.

Reply:
If you really don't want to fetch it twice, you could also add an optional metadata parameter (default null) and only fetch if it's null.